### PR TITLE
Bug1592290 5.7

### PR DESCRIPTION
**Pull request by laurynas-biveinis**
http://jenkins.percona.com/job/mysql-5.7-param/233/
http://jenkins.percona.com/job/mysql-5.7-param/248/

### CODE REVIEW

File: storage/innobase/btr/btr0sea.cc [hunk 1]
```diff
@@ -101,6 +108,32 @@ btr_search_get_n_fields(
 	return(n_fields + (n_bytes > 0 ? 1 : 0));
 }
```
Comment (percona-ysorokin) [inline storage/innobase/btr/btr0sea.cc:110]: May be better to use a macro turning into nothing in release mode
something like

``` c++
#ifdef UNIV_DEBUG
  #define btr_search_mem_accounting_validate() btr_search_mem_accounting_validate_impl()
#else
  #define btr_search_mem_accounting_validate()
#endif
static void btr_search_mem_accounting_validate_impl(void) { ... }
```

File: storage/innobase/btr/btr0sea.cc [hunk 2]
```diff
@@ -1720,7 +1805,15 @@ btr_search_update_hash_on_delete(btr_cur_t* cursor)
 	if (block->index) {
 		ut_a(block->index == index);
 
+		ulint old_ht_heap_size = mem_heap_get_size(table->heap);
 		if (ha_search_and_delete_if_found(table, fold, rec)) {
```
Comment (percona-ysorokin) [inline storage/innobase/btr/btr0sea.cc:1811]: The same

``` c++
ut_d(os_rmb);
ut_ad(btr_search_sys_variable_mem >= old_ht_heap_size);
```

as in `btr_search_drop_page_hash_index()` ?

**Comment by percona-ysorokin** [comment]
LGTM, comments are minor

File: storage/innobase/btr/btr0sea.cc (hunk 1 above)
Comment (laurynas-biveinis) [inline storage/innobase/btr/btr0sea.cc:110]: This does not look very clean, the function is static to this file and has empty body in release build. The compiler should be able to optimise the call away completely.

**Comment by laurynas-biveinis** [comment]
http://jenkins.percona.com/job/mysql-5.7-param/250/

**Pull request merged**